app/services/recovery_code_generator.rb

```diff
@@ -10,12 +10,33 @@ def create
     raw_recovery_code
   end
 
+  def verify(plaintext_code)
+    uak = user_access_key(plaintext_code)
+    encryption_key, encrypted_code = user.recovery_code.split(Pii::Encryptor::DELIMITER)
+    begin
+      key_maker.unlock(uak, encryption_key)
+    rescue Pii::EncryptionError => _err
+      return false
+    end
+    Devise.secure_compare(encrypted_code, uak.encrypted_password)
+  end
+
   private
 
   attr_reader :length, :user
 
+  def key_maker
+    EncryptedKeyMaker.new
```

Comment:
good point. I was originally trying to get around a reek warning about too many lines in a method, but the key maker is stateless so now I just instantiate one in `initialize`.